include/engine/hint.hpp

```diff
@@ -57,16 +58,31 @@ struct Hint
                  const datafacade::BaseDataFacade &facade) const;
 
     std::string ToBase64() const;
-    static Hint FromBase64(const std::string &base64Hint);
+    static SegmentHint FromBase64(const std::string &base64Hint);
 
-    friend bool operator==(const Hint &, const Hint &);
-    friend std::ostream &operator<<(std::ostream &, const Hint &);
+    friend bool operator==(const SegmentHint &, const SegmentHint &);
+    friend bool operator!=(const SegmentHint &, const SegmentHint &);
+    friend std::ostream &operator<<(std::ostream &, const SegmentHint &);
 };
 
-static_assert(sizeof(Hint) == 80 + 4, "Hint is bigger than expected");
-constexpr std::size_t ENCODED_HINT_SIZE = 112;
-static_assert(ENCODED_HINT_SIZE / 4 * 3 >= sizeof(Hint),
-              "ENCODED_HINT_SIZE does not match size of Hint");
+// Hint represents the suggested segment positions that could be used
+// as the waypoint for a given input location
+struct Hint
```

Comment:
Not sure if it is not breaking change too - as I understand API of Hint class is part of public API too, BUT tbh I don’t think it is a big problem: even though someone uses it for something(I doubt it) it will be quite easy to migrate.

Reply:
Sorry didn’t notice comment https://github.com/Project-OSRM/osrm-backend/pull/5953#issuecomment-1222933925